demo/test/components/previews/primer/underline_nav_component_preview.rb

```diff
@@ -0,0 +1,20 @@
+module Primer
+  class UnderlineNavComponentPreview < ViewComponent::Preview
+    def default
+      render(Primer::UnderlineNavComponent.new(with_panel: true)) do |c|
+        c.tab(selected: true) do |t|
+          t.panel { "Panel 1" }
+          "Tab 1"
```

Comment:
How do you feel about mixing slots with a block like this? What if we instead did something like:

```suggestion
          t.panel { "Panel 1" }
          t.label { "Tab 1" } 
```

Reply:
I wanted to avoid another sub-slot, since most of the use cases **aren't** using `tab-container`, so 90% of the time we won't even call `t.panel`

Reply:
@joelhawksley I'm going to merge as-is and work in the improvements we discussed in the follow-up PR https://github.com/primer/view_components/pull/393